src/ngMock/angular-mocks.js

```diff
@@ -1378,9 +1378,13 @@ function createHttpBackendMock($rootScope, $timeout, $delegate, $browser) {
     function wrapResponse(wrapped) {
       if (!$browser && timeout) {
         if (timeout.then) {
-          timeout.then(handleTimeout);
+          timeout.then(function() {
+            handlePrematureEnd(timeout.$$timeoutId >= 0 ? 'timeout' : 'abort');
```

Comment:
Why not keep in-line with the actual implementation and use `isDefined(timeout.$$timeoutId)`? Does it make a difference?